### PR TITLE
tag: Disallow conflicting file tags and use functions to validate

### PR DESCRIPTION
**Pull request by Amorymeltzer**
1st commit disallows some conflicting file tags, as requested at [WT:TW](https://en.wikipedia.org/wiki/Wikipedia_talk:Twinkle/Archive_42#Mutually_exclusive_tags).  There are probably more we could do, this covers:
- Only one of {{Bad GIF}}, {{Bad JPEG}}, {{Bad SVG}}, and {{Bad format}} is used
- Only one of {{Should be PNG}}, {{Should be SVG}}, and {{Should be text}} is used
- Only one of {{Bad SVG}} and {{Vector version available}} is used
- Only one of {{Bad JPEG}} and {{Overcompressed JPEG}} is used
- Only one of {{PNG version available}} and {{Vector version available}} is used
- The correct extension is used for Bad GIF/JPEG/SVG and Should be PNG/SVG
- Only allows JPEGs to be tagged with {{Overcompressed JPEG}}
- Only allows SVGs to be tagged with {{Bad trace}} and {{Bad font}}

2nd commit uses `checkIncompatible` and `checkParameter` to make the repetitive validation measures more straightforward.

### CODE REVIEW

File: modules/friendlytag.js [hunk 1]
```diff
@@ -2009,47 +2035,97 @@ Twinkle.tag.callback.evaluate = function friendlytagCallbackEvaluate(e) {
...
+			if (checkIncompatible(['Bad JPEG', 'Overcompressed JPEG'])) {
+				return;
+			}
+			if (checkIncompatible(['PNG version available', 'Vector version available'])) {
+				return;
+			}
+
+			var extension;
+			if ((extension = Morebits.pageNameNorm.match(/\.(\w+)$/)) !== null) {
+				extension = extension[1];
```
Comment (Amorymeltzer) [inline modules/friendlytag.js:2072]: Per a [discussion with the requester](https://en.wikipedia.org/wiki/User_talk:Smjg#Twinkle_pull_request_for_your_file_incompatibility_checks), it should be possible to use MIME types for jpg/gif/png, with a fallback to this for svgs.  Something stupid like `((extension = $('.mime-type').text()) && extension.split(/\//)[1]) || ((extension = Morebits.pageNameNorm.match(/\.(\w+)$/)) !== null) && (extension = extension[1])` works 😆

Comment (siddharthvp) [reply modules/friendlytag.js:2072]: To get the extension, we can use `mw.Title.newFromText(Morebits.pageNameNorm).getExtension()` as well.

Comment (Amorymeltzer) [reply modules/friendlytag.js]: Done

**Comment by Amorymeltzer** [comment]
Unrelated, but woo, 1000 issues/PRs!

**Pull request merged**